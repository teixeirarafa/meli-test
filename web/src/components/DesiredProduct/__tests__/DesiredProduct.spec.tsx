/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { render } from "@testing-library/react";
import DesiredProduct from "..";

describe("DesiredProduct Component", () => {
  it("should render DesiredProduct with informations", () => {
    const productInformations = {
      condition: "Nuevo",
      name: "Rafael",
      sales: 10,
      value: "1000",
      description: "description",
      onPurchaseClick: () => ({}),
    };

    const app = render(
      <DesiredProduct symbolCurrency="$" picture="" {...productInformations} />,
    );

    expect(app).toMatchSnapshot();
  });
});
