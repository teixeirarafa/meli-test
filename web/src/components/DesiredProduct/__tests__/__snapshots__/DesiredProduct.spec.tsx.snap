// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesiredProduct Component should render DesiredProduct with informations 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="desired-product"
      >
        <figure
          class="product__picture product__picture--large"
        >
          <img
            alt="Rafael"
            src=""
          />
        </figure>
        <section
          class="product__informations product__informations--medium"
        >
          <span
            class="product-informations__sales"
          >
            Nuevo
             - 
            10
             vendidos
          </span>
          <div
            class="product__general-informations"
          >
            <div
              class="product__price"
            >
              <span
                class="price__symbol margin-right--5"
              >
                $
              </span>
              <span
                class="price__value"
              >
                1000
              </span>
            </div>
          </div>
          <h1
            class="product__name margin-top--16"
          >
            Rafael
          </h1>
          <button
            class="btn btn--rounded btn--full bg--blue color--white border--none margin-top--32"
            type="button"
          >
            Comprar
          </button>
        </section>
        <div
          class="product__description"
        >
          <h2>
            Descripción del producto
          </h2>
          <p
            class="margin-top--32"
          >
            description
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="desired-product"
    >
      <figure
        class="product__picture product__picture--large"
      >
        <img
          alt="Rafael"
          src=""
        />
      </figure>
      <section
        class="product__informations product__informations--medium"
      >
        <span
          class="product-informations__sales"
        >
          Nuevo
           - 
          10
           vendidos
        </span>
        <div
          class="product__general-informations"
        >
          <div
            class="product__price"
          >
            <span
              class="price__symbol margin-right--5"
            >
              $
            </span>
            <span
              class="price__value"
            >
              1000
            </span>
          </div>
        </div>
        <h1
          class="product__name margin-top--16"
        >
          Rafael
        </h1>
        <button
          class="btn btn--rounded btn--full bg--blue color--white border--none margin-top--32"
          type="button"
        >
          Comprar
        </button>
      </section>
      <div
        class="product__description"
      >
        <h2>
          Descripción del producto
        </h2>
        <p
          class="margin-top--32"
        >
          description
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
